### PR TITLE
Move vector embeddings to HANA section

### PR DESCRIPTION
**Pull request by MattSchur**

### CODE REVIEW

File: java/cds-data.md [hunk 1]
```diff
@@ -280,11 +280,7 @@ Avoid cyclic relationships between CdsData objects when using toJson.
...
-
-Typically vector embeddings are computed using models tailored to a specific use case, like large language models (LLMs) for text, or convolutional neural networks (CNNs) for images. The dimensionality of the vector embedding space depends on the chosen model. Unified LLM consumption across different vendors and open source models is provided via the [SAP Generative AI Hub](https://community.sap.com/t5/technology-blogs-by-sap/how-sap-s-generative-ai-hub-facilitates-embedded-trustworthy-and-reliable/ba-p/13596153).
-
-In CDS, such vector embeddings are stored in elements of type `cds.Vector`:
+In CDS [vector embeddings](../guides/databases-hana#vector-embeddings) are stored in elements of type `cds.Vector`:
```
Comment (johannes-vogel) [inline java/cds-data.md:283]: Should we move the cds definition also to the common section?

File: guides/databases-hana.md [hunk 2]
```diff
@@ -235,6 +235,15 @@ See the [Deploying to Cloud Foundry](deployment/) guide for information about ho
...
+
+Vector embeddings are numerical representations that capture features and inherent semantics of unstructured data - such as text, images, or audio. They facilitate tasks like similarity search, recommendations, and Retrieval Augmented Generation (RAG). This improves the results of generative AI (GenAI) by augmenting prompts with relevant data retrieved from a vector datastore such as the [SAP HANA Cloud Vector Engine](https://community.sap.com/t5/technology-blogs-by-sap/sap-hana-cloud-s-vector-engine-announcement/ba-p/13577010).
+
+Typically vector embeddings are computed using models tailored to a specific use case, like large language models (LLMs) for text, or convolutional neural networks (CNNs) for images. The dimensionality of the vector embedding space depends on the chosen model. Unified LLM consumption across different vendors and open source models is provided via the [SAP Generative AI Hub](https://community.sap.com/t5/technology-blogs-by-sap/how-sap-s-generative-ai-hub-facilitates-embedded-trustworthy-and-reliable/ba-p/13596153).
+
+[Learn more about Vector Embeddings in CAP Java](../java/cds-data#vector-embeddings) {.learn-more}
```
Comment (johannes-vogel) [inline guides/databases-hana.md:244]: 👍 I'll create a similar link for Node.js

File: java/cds-data.md (hunk 1 above)
Comment (MattSchur) [reply java/cds-data.md:283]: I would like to keep the CDS model in the Java section for better readability of the Java examples. I added a link to the build-in-types section. We could also duplicate the CDS model example

Comment (agoerler) [reply java/cds-data.md:283]: We could have the definition in both places.

**Comment by johannes-vogel** [comment]
@MattSchur @agoerler With [5b2cda2](https://github.com/cap-js/docs/pull/768/commits/5b2cda299f37b57fb7fa1bf4f965dc31a6a6a487) I propose to have also a runtime sample within the guide. From my perspective that rounds up the section. More details could be found/mentioned in the runtime specific docs.

What do you think?

File: guides/databases-hana.md [hunk 3]
```diff
@@ -235,6 +235,45 @@ See the [Deploying to Cloud Foundry](deployment/) guide for information about ho
...
+```
+
+```java [Java]
+//  Compute vector embedding, e.g. via LangChain4j
+float[] embedding = llm.embed(text).content().vector();
+
+CqnSelect query = Select.from(BOOKS).where(b -> // [!code focus]
+  CQL.cosineSimilarity(b.embedding(), CQL.vector(embedding)).gt(0.9)) // [!code focus]
+
+Result relatedBooks = service.run(query); // [!code focus]
```
Comment (MattSchur) [inline guides/databases-hana.md:270]:
```suggestion
// Vector embedding of text, e.g. from SAP GenAI Hub or via LangChain4j
float[] embedding = llm.embed(text).content().vector();

CqnSelect query = Select.from(BOOKS).where(b -> 
  CQL.cosineSimilarity(b.embedding(), CQL.vector(embedding)).gt(0.9))

Result relatedBooks = service.run(query);
```

File: guides/databases-hana.md [hunk 4]
```diff
@@ -235,6 +235,45 @@ See the [Deploying to Cloud Foundry](deployment/) guide for information about ho
...
+In CAP, vector embeddings are stored in elements of type [cds.Vector](../cds/types.md#built-in-types):
+
+```cds
+entity Books : cuid { // [!code focus]
+  title         : String(111);
+  embedding     : Vector(1536); // vector space w/ 1536 dimensions // [!code focus]
+} // [!code focus]
+```
+
+At runtime, you can use it leveraging the Query API:
```
Comment (MattSchur) [inline guides/databases-hana.md:253]: or
```suggestion
At runtime, you can use them in queries:
```

**Pull request merged**